### ISSUE TITLE
[Responsive Navigation] "Hidden" elements in example take a while to hide...

### ISSUE THREAD

**Issue by trinode** [zurb/foundation-sites#7735]
Paste the example which shows the top bar or titlebar responsively, refresh, browse or something, watch BOTH appear momentarily (for mobile or desktop view sizes, then the correct one is displayed only after an anoying flicker of both).

Adding ```hide-for-medium``` to the title-bar class element fixes the issue for both showing at desktop sizes:-

```
<div class="title-bar hide-for-medium" data-responsive-toggle="example-menu" data-hide-for="medium">
  <button class="menu-icon" type="button" data-toggle></button>
  <div class="title-bar-title">Menu</div>
</div>
```

but adding ```show-for-medium``` to the top-bar class results in it not being able to be shown by the toggle at mobile sizes.

Is the example poor, or is it not possible in F6S to hide elements via CSS (like I tried, and half succeeded instead of waiting for JS to run) and show them by JS with the current responsive components?

Any pointers appreciated.

**Comment by trinode** [comment]
Um Same issue as #7083 - closing in favour of that.
